From 7dd22c493b5ba6d7e4d731f6a7cbf0b28251d860 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Thu, 29 Jun 2017 00:24:51 +0000
Subject: [PATCH] MANTA-3335 Muskie doesn't return after invoking callback with
 NotEnoughSpaceError Reviewed by: Jordan Hendricks
 <jordan.hendricks@joyent.com> Approved by: Jordan Hendricks
 <jordan.hendricks@joyent.com>

---
 lib/picker.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/picker.js b/lib/picker.js
index b2e797c..96f0f46 100644
--- a/lib/picker.js
+++ b/lib/picker.js
@@ -473,6 +473,7 @@ Picker.prototype.choose = function choose(opts, cb) {
 
         if (!sharks.length && (!tuple || tuple.length < replicas)) {
             cb(new NotEnoughSpaceError(size));
+            return;
         } else if (tuple && this.multiDC && replicas > 1) {
             var _dcs = tuple.map(function (s) {
                 return (s.datacenter);
-- 
2.21.0

